src/chrome/utils/typedOperators.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+/**
+ * These file includes functions normally implemented by JavaScript operators. The reason we use these functions
+ * is to have better control over the typing of the functions, and use that to help reduce the chance of introducing errors
+ * e.g.: We on purpose do not allowe isDefined() to be called for strings, because there is a fair chance that we meant to use isNotEmpty instead
+ */
+
+/**
+ * Returns whether the object or number is an actual object or number (and is not undefined)
+ */
+export function isDefined<T extends object | number>(objectOrUndefined: T | undefined): objectOrUndefined is T {
```

Comment:
Why `object | number`? Does that include string? Do you really need the type constraint?

Reply:
Oh I read the comment, got it